src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -404,26 +490,31 @@ private boolean verifyStatus(String chainTaskId, String walletAddress, Replicate
         }
     }
 
-    private boolean updateReplicateWeight(String chainTaskId, Replicate replicate) {
+    private boolean isWorkerWeightPresent(String chainTaskId,
+                                          Replicate replicate,
+                                          UpdateReplicateStatusArgs updateReplicateStatusArgs) {
         String walletAddress = replicate.getWalletAddress();
-        int workerWeight = iexecHubService.getWorkerWeight(walletAddress);
+        int workerWeight = updateReplicateStatusArgs.getWorkerWeight();
         if (workerWeight == 0) {
             log.error("Failed to get worker weight [chainTaskId:{}, workerWallet:{}]",
                     chainTaskId, walletAddress);
             return false;
         }
+        return true;
+    }
 
-        Optional<ChainContribution> chainContribution =
-                iexecHubService.getChainContribution(chainTaskId, walletAddress);
+    private boolean isChainContributionPresent(String chainTaskId,
+                                               Replicate replicate,
+                                               UpdateReplicateStatusArgs updateReplicateStatusArgs) {
+        ChainContribution chainContribution = updateReplicateStatusArgs.getChainContribution();
```

Comment:
Since we're at it, isn't it enough to pass `chainContribution` as an argument to this method instead of `UpdateReplicateStatusArgs`?

Reply:
Same here, we should rephrase the method name to be clearer :)